core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareHandler.java

```diff
@@ -124,6 +124,7 @@ protected CqlPrepareHandler(
           try {
             if (t instanceof CancellationException) {
               cancelTimeout();
+              context.getRequestThrottler().signalCancel(this);
```

Comment:
One thought I immediately had was, "is it possible that we might signal to the throttler twice for the same request?".

It doesn't appear to be possible from what I can tell.   It looks like `CompletableFuture.cancel(..)` cannot cancel a completed exception, so this should be fine.